File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int , int>mp;
        for(int i = 0; i<n; i++){
            mp[nums[i]] = i;
        }
        vector<int>ans;
        for(int i = 0; i<n; i++){
            if(mp.count(target - nums[i]) and i != mp[target - nums[i]]){
                ans.push_back(i);
                ans.push_back(mp[target - nums[i]]);
                return ans;
            }
        }
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const mp: Map<number , number> = new Map();
    for(let i = 0; i<nums.length; i++){
        const complement = target - nums[i];
        if(mp.has(complement)){
            return [i , mp.get(complement)];
        }
        mp.set(nums[i] , i);
    }
    return [];
};